executor/api/rest/utils.go

```diff
@@ -14,3 +24,83 @@ func ExtractRouteAsJsonArray(msg []byte) ([]int, error) {
 		return nil, err
 	}
 }
+
+func embedSeldonDeploymentValuesInJson(namespace string, sdepName string, openapiInterface *interface{}) error {
+	jsonFormatError := errors.New("Incorrect format for OpenAPI JSON file")
+
+	replacer := strings.NewReplacer(
+		"{namespace}", namespace,
+		"{deployment}", sdepName)
+
+	// Ensure json is a map before performing actions
+	if openapiJson, ok := (*openapiInterface).(map[string]interface{}); ok {
+		// Remove the servers element to ensure it uses the same URL
+		delete(openapiJson, "servers")
+
+		// Get the paths key value to remove the parameters from each of the URLs
+		if pathsJson, ok := openapiJson["paths"].(map[string]interface{}); ok {
+			// Delete the parameters field from the predictions path
+			if openapiPredPathJson, ok := pathsJson[openapiPredPath].(map[string]interface{}); ok {
+				if openapiPredPathPostJson, ok := openapiPredPathJson["post"].(map[string]interface{}); ok {
+					delete(openapiPredPathPostJson, "parameters")
+				} else {
+					return jsonFormatError
+				}
+			} else {
+				return jsonFormatError
+			}
+
+			// Rename the predictions path to use the namespace and deploymentName instead of placeholder values
+			openapiPredPathReplaced := replacer.Replace(openapiPredPath)
+			pathsJson[openapiPredPathReplaced] = pathsJson[openapiPredPath]
+			delete(pathsJson, openapiPredPath)
+
+			// Delete the parameters field from the feedback path
+			if openapiFeedbackPathJson, ok := pathsJson[openapiFeedbackPath].(map[string]interface{}); ok {
+				if openapiFeedbackPathPostJson, ok := openapiFeedbackPathJson["post"].(map[string]interface{}); ok {
+					delete(openapiFeedbackPathPostJson, "parameters")
+				} else {
+					return jsonFormatError
+				}
+			} else {
+				return jsonFormatError
+			}
+
+			// Rename the predictions path to use the namespace and deploymentName instead of placeholder values
+			openapiFeedbackPathReplaced := replacer.Replace(openapiFeedbackPath)
+			pathsJson[openapiFeedbackPathReplaced] = pathsJson[openapiFeedbackPath]
+			delete(pathsJson, openapiFeedbackPath)
+		} else {
+			return jsonFormatError
+		}
+
+	} else {
+		return jsonFormatError
+	}
+
+	return nil
+}
+
+func EmbedSeldonDeploymentValuesInSwaggerFile(namespace string, sdepName string) error {
+	openapiInputBytes, err := ioutil.ReadFile(openapiFilePath)
+	if err != nil {
+		return err
+	}
+	var openapiInterface interface{}
+	if err := json.Unmarshal(openapiInputBytes, &openapiInterface); err != nil {
+		return err
+	}
+
+	if err := embedSeldonDeploymentValuesInJson(namespace, sdepName, &openapiInterface); err != nil {
+		return err
+	}
+
+	// We use MarshalIndent so that the output can be humanly visible and indented
+	openapiOutputBytes, err := json.MarshalIndent(openapiInterface, "", "    ")
+
+	if err := ioutil.WriteFile(openapiFilePath, openapiOutputBytes, 0644); err != nil {
```

Comment:
If this happens at runtime, how does the `/api/v1.0/doc/seldon.json` endpoint find the updated schema (with the injected deployment details)? Is `openapiFilePath` relative to the execution path?

Reply:
Yes, correct, as abve the executor becomes a single executable file, so the folder lives exacly right next to it.